perllib/FixMyStreet/App/Controller/Waste.pm

```diff
@@ -587,6 +587,16 @@ sub csc_payment_failed : Path('csc_payment_failed') : Args(0) {
 sub property : Chained('/') : PathPart('waste') : CaptureArgs(1) {
     my ($self, $c, $id) = @_;
 
+    # Some pages chained off /waste/property require user to be logged in, so
+    # perform /auth redirect before we start looking up the property.
+    # These pages include:
+    # /waste/<property_id>/bulky/cancel/<report_id>
+    if ( $c->req->path =~ /\/bulky\/cancel/
```

Comment:
Think this is okay place to put it! Would add to the comment that it's the asynchronous property lookup that makes this necessary as the redirect to auth then doesn't work after that.
And then this doesn't catch the small items path, so might be better to have `$c->action eq 'waste/bulky/cancel' || $c->action eq 'waste/bulky/cancel_small'` (doesn't need to worry about the URLs then either) (I think those are the right actions,but needs checking)